pkg/pillar/cmd/zedrouter/appcontainer.go

```diff
@@ -120,10 +134,81 @@ func checkAppStopStatsCollect(ctx *zedrouterContext) (map[string]interface{}, bo
 	return items, false
 }
 
-func appContainerGetStats(ipAddr net.IP) (types.AppContainerMetrics, error) {
+func getAppContainerStats(status types.AppNetworkStatus, cli *client.Client, containers []apitypes.Container) types.AppContainerMetrics {
 	var acMetrics types.AppContainerMetrics
-	// XXX collect container stats for each container with the docker API endpoint ipAddr
-	return acMetrics, nil
+
+	for _, container := range containers {
+		// the main purpose of Inspect is to obtain the container start time for CPU stats
+		cjson, err := cli.ContainerInspect(context.Background(), container.ID)
+		if err != nil {
+			log.Errorf("getAppContainerStats: container inspect for %s, error %v", container.ID, err)
+			continue
+		}
+		startTime, _ := time.Parse(time.RFC3339Nano, cjson.State.StartedAt)
+
+		stats, err := cli.ContainerStats(context.Background(), container.ID, false)
+		if err != nil {
+			log.Errorf("getAppContainerStats: container stats for %s, error %v\n", container.Names[0], err)
+			continue
+		}
+
+		acStats, err := processAppContainerStats(stats, container, startTime)
+		if err != nil {
+			log.Errorf("getAppContainerStats: process stats for %s, error %v\n", container.Names[0], err)
+			continue
+		}
+		log.Debugf("getAppContainerStats: container stats %v", acStats)
+		acMetrics.StatsList = append(acMetrics.StatsList, acStats)
+	}
+
+	return acMetrics
+}
+
+func processAppContainerStats(stats apitypes.ContainerStats, container apitypes.Container, startTime time.Time) (types.AppContainerStats, error) {
+	var acStats types.AppContainerStats
+	var v *apitypes.StatsJSON
+
+	dec := json.NewDecoder(stats.Body)
+	err := dec.Decode(&v)
+	if err != nil {
+		return acStats, err
+	}
+
+	acStats.ContainerName = strings.Trim(container.Names[0], "/")
+	acStats.Status = container.Status
+
+	acStats.Pids = uint32(v.PidsStats.Current)
+
+	// Container CPU stats, convert from nano-seconds to seconds
```

Comment:
This might not survive a change to the go packages. It merely happens that time.Time is nanoseconds and that might change in some future. But I suspect that the input from apitypes.ContainerStats is actually defined to be in nanoseconds. Hence shouldn't assume that time.Second will always be 10^9.

Reply:
ok, i'll define it explicitly